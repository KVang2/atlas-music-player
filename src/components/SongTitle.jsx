import React from 'react';

export default function SongTitle() {
    return (
        <div className="flex flex-col text-start p-4">
            <p className="font-bold text-text">Tidal Drift</p>
            <p className="text-text">Echoes of the Sea</p>
        </div>
    );
}