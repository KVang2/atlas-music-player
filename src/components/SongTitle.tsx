interface SongTitleProps {
    title: string;
    artist: string;
}

export default function SongTitle({ title, artist }: SongTitleProps) {
    return (
        <div className="flex flex-col text-start p-4 dark:">
            <p className="font-bold text-text">{title}</p>
            <p className="font-light text-text">{artist}</p>
        </div>
    );
}