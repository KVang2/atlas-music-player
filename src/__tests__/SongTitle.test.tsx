import SongTitle from "../components/SongTitle";
import { render, screen } from "@testing-library/react";
import { describe, test, expect } from "vitest";

describe ("SongTitle", () => {
    const title = "Painted in Blue";
    const artist = "Soul Canvas";

    test("render correct title", () => {
        render(<SongTitle title={title} artist={artist} />);

        const titles = screen.getByText(title);
        expect(titles).toBeInTheDocument();
        expect(titles).toHaveTextContent(title);
    });

    test("renders correct artist name", () => {
        render(<SongTitle title={title} artist={artist} />);

        const artistElement = screen.getByText(new RegExp(artist, "i"));
        expect(artistElement).toBeInTheDocument();
        expect(artistElement).toHaveTextContent(artist);
    });

    test("matchs snapshot", () => {
        const { container } = render(<SongTitle title={title} artist={artist} />);
        expect(container).toMatchSnapshot();
    });
});