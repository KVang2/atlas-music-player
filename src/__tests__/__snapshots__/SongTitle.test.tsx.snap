// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle > matchs snapshot 1`] = `
<div>
  <div
    class="flex flex-col text-start p-4 dark:"
  >
    <p
      class="font-bold text-text"
    >
      Painted in Blue
    </p>
    <p
      class="font-light text-text"
    >
      Soul Canvas
    </p>
  </div>
</div>
`;
